Classes with @Funq methods fail in constructor injection for normal scoped bean

### Describe the bug

When class that is normal scoped bean has method annotated with `@Funq` and uses constructor injection, an exception is thrown. I looked into this and the reason is that when (and only when!) class does extend SomeClass and there is no default constructor without args, Funqy generates it for us here https://github.com/quarkusio/quarkus/blob/b9d7536effc07f08f119eea292673da6e0826057/extensions/funqy/funqy-server-common/deployment/src/main/java/io/quarkus/funqy/deployment/FunctionScannerBuildStep.java#L176

but Quarkus Arc does too for normal scoped beans unless a bean class extends a class that does not declare a no-args constructor.

### Expected behavior

Constructor injection for normal scoped beans works just fine.

### Actual behavior

An exception is thrown: `Exception in thread "main" java.lang.reflect.InvocationTargetException`
`Caused by: java.lang.ClassFormatError: Duplicate method name "<init>" with signature "()V" in class file io/quarkus/funqy/deployment/DependencyInjectionTest$HiFunction`

### How to Reproduce?

1. `git clone https://github.com/michalvavrik/funqy-knative-events-constructor-injection-reproducer.git`
2. `mvn clean verify`


### Output of `uname -a` or `ver`

Linux fedora 5.19.4-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Aug 25 17:42:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.3" 2022-04-19

### GraalVM version (if different from Java)

GraalVM CE 22.1.0

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

_No response_

--- a/extensions/funqy/funqy-server-common/deployment/pom.xml
+++ b/extensions/funqy/funqy-server-common/deployment/pom.xml
@@ -25,6 +25,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-arc-deployment</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5-internal</artifactId>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>

--- a/extensions/funqy/funqy-server-common/deployment/src/main/java/io/quarkus/funqy/deployment/FunctionScannerBuildStep.java
+++ b/extensions/funqy/funqy-server-common/deployment/src/main/java/io/quarkus/funqy/deployment/FunctionScannerBuildStep.java
@@ -172,6 +172,12 @@ public class FunctionScannerBuildStep {
             Set<ClassInfo> withoutDefaultCtor) {
 
         for (ClassInfo classInfo : withoutDefaultCtor) {
+            // don't generate constructor for normal scoped beans as the Quarkus Arc does that for us
+            final BuiltinScope scope = BuiltinScope.from(classInfo);
+            if (scope != null && scope.getInfo().isNormal()) {
+                continue;
+            }
+
             // keep it super simple - only generate default constructor is the object is a direct descendant of Object
             if (!(classInfo.superClassType() != null && classInfo.superClassType().name().equals(DotNames.OBJECT))) {
                 return;

--- a/extensions/funqy/funqy-server-common/deployment/src/test/java/io/quarkus/funqy/deployment/DependencyInjectionTest.java
+++ b/extensions/funqy/funqy-server-common/deployment/src/test/java/io/quarkus/funqy/deployment/DependencyInjectionTest.java
@@ -1,0 +1,127 @@
+package io.quarkus.funqy.deployment;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.funqy.Funq;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DependencyInjectionTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest test = new QuarkusUnitTest()
+            .withApplicationRoot(
+                    (jar) -> jar.addClasses(GreetingFunction.class, GreetingService.class, FarewellFunction.class,
+                            HiFunction.class, ByeFunction.class));
+
+    @Inject
+    GreetingFunction greetingFunction;
+
+    @Inject
+    FarewellFunction farewellFunction;
+
+    @Inject
+    ByeFunction byeFunction;
+
+    @Inject
+    HiFunction hiFunction;
+
+    @Test
+    public void testFieldInjection() {
+        // bean with a pseudo-scope
+        Assertions.assertEquals("Hello World!", greetingFunction.greeting());
+        // normal scoped bean
+        Assertions.assertEquals("Bye!", byeFunction.bye());
+    }
+
+    @Test
+    public void testConstructorInjection() {
+        // bean with a pseudo-scope
+        Assertions.assertEquals("Goodbye!", farewellFunction.farewell());
+        // normal scoped bean
+        Assertions.assertEquals("Hi!", hiFunction.hi());
+    }
+
+    public static class GreetingFunction {
+
+        @Inject
+        GreetingService greetingService;
+
+        @Funq
+        String greeting() {
+            return greetingService.sayHello();
+        }
+
+    }
+
+    public static class FarewellFunction {
+
+        private final GreetingService greetingService;
+
+        public FarewellFunction(GreetingService greetingService) {
+            this.greetingService = greetingService;
+        }
+
+        @Funq
+        String farewell() {
+            return greetingService.sayGoodbye();
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class ByeFunction {
+
+        @Inject
+        GreetingService greetingService;
+
+        @Funq
+        String bye() {
+            return greetingService.sayBye();
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class HiFunction {
+
+        private final GreetingService greetingService;
+
+        public HiFunction(GreetingService greetingService) {
+            this.greetingService = greetingService;
+        }
+
+        @Funq
+        String hi() {
+            return greetingService.sayHi();
+        }
+
+    }
+
+    @Singleton
+    public static class GreetingService {
+
+        public String sayHello() {
+            return "Hello World!";
+        }
+
+        public String sayGoodbye() {
+            return "Goodbye!";
+        }
+
+        public String sayHi() {
+            return "Hi!";
+        }
+
+        public String sayBye() {
+            return "Bye!";
+        }
+
+    }
+
+}

